### PR TITLE
A4A: Implement Pressable usage section on the new Hosting page v3.

### PR DESCRIPTION
**Pull request by jkguidaven**
This PR includes the usage section for Pressable plans in the new hosting page v3. It displays only if the current user has a Pressable plan purchased through A4A.

**NOTE: This PR will update the production appearance, but it's acceptable since it does not alter any functionality, just the presentation.**

Screenshot 2024-12-16 at 6 06 17 PM

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1551

## Proposed Changes

* Include the usage section in the Pressable section only if the current agency has a plan. Additionally, the usage will not be displayed in referral mode or if the current user has a Pressable plan that was not purchased from A4A.
* **Additional**: We have now moved the logic for loading the section title from the Plan section scope. This way, we can load different text based on specific contexts (e.g. existing plans). 
* **Additional**: Not part of the main goal but a minor fix that adds a handler for the 'Request demo' button click.

## Why are these changes being made?

* This is part of the In-product hosting page v3 project.

## Testing Instructions

* Use the A4A live link and go to the `/marketplace/hosting/pressable` section.
* Make sure you purchase a Pressable plan and have some usage.
* Confirm that in the page, you see the usage stats with correct information.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/marketplace/hosting-overview-v3/hosting-content/premier-agency-hosting/pressable-plan-section/index.tsx [hunk 1]
```diff
@@ -276,8 +298,11 @@ export default function PressablePlanSection( {
 				heading={ translate( 'Schedule a demo' ) }
 				cta={ {
 					label: translate( 'Schedule a demo' ),
-					onClick: () => {},
+					onClick: onScheduleDemo,
+					href: PRESSABLE_CONTACT_LINK,
+					target: '_blank',
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/hosting-overview-v3/hosting-content/premier-agency-hosting/pressable-plan-section/index.tsx:303]: Not part of the main goal but a minor fix that adds a handler for the 'Request demo' button click.

File: client/a8c-for-agencies/sections/marketplace/hosting-overview-v3/hosting-content/index.tsx [hunk 2]
```diff
@@ -7,54 +6,34 @@ import EnterpriseAgencyHosting from '../../hosting-overview/hosting-v2/enterpris
...
-	}, [ isReferMode, onAddToCart, section, translate ] );
+	}, [ isReferMode, onAddToCart, section ] );
 
-	return (
-		<div className="hosting-v3__content">
-			<h2 className="hosting-v3__content-header">{ title }</h2>
-			{ content }
-		</div>
-	);
+	return <div className="hosting-v3__content">{ content }</div>;
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/hosting-overview-v3/hosting-content/index.tsx:38]: We have now moved the logic for loading the section title from the Plan section scope. This way, we can load different text based on specific contexts (e.g. existing plans).

**Review by cleacos — APPROVED** [review]
The code looks good. It works as expected.

LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17065730

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17065730).

**Pull request merged**